### PR TITLE
override the default translation table

### LINKED ISSUE #95: allow setting of alternate translation codes per organism
About testing alternate codes (translation tables) @selewis asked: 
During the testing was there testing for the alternate protein coding
function? (for those weird creatures that don't use the same translation
code as most others).

### PR DESCRIPTION
**Pull request by nathandunn**
Fixes #95  for having a per-organism translation table.

@deepakunni3  Can you review when you have a moment.

### CODE REVIEW

File: grails-app/controllers/org/bbop/apollo/AnnotationEditorController.groovy [hunk 1]
```diff
@@ -140,12 +141,25 @@ class AnnotationEditorController extends AbstractApolloController implements Ann
...
+        TranslationTable translationTable
+        log.debug "has organism ${organism}"
+        // use the over-wridden one
+        if(organism?.nonDefaultTranslationTable){
+            log.debug "overriding default translation table for ${organism.commonName} with ${organism.nonDefaultTranslationTable}"
+            translationTable = SequenceTranslationHandler.getTranslationTableForGeneticCode(organism.nonDefaultTranslationTable)
+        }
+            // just use the default
+        else{
+            log.deub "using the default translation table"
```
Comment (deepakunni3) [inline grails-app/controllers/org/bbop/apollo/AnnotationEditorController.groovy]: `log.deub` - surprisingly this line is not throwing any errors.

Comment (deepakunni3) [inline grails-app/controllers/org/bbop/apollo/AnnotationEditorController.groovy]: actually it does:
```
No signature of method: org.apache.commons.logging.impl.SLF4JLog.deub() is applicable for argument types: (java.lang.String) values: [using the default translation table]
Possible solutions: debug(java.lang.Object), dump(), debug(java.lang.Object, java.lang.Throwable), getAt(java.lang.String), wait(), any(). Stacktrace follows:
Message: No signature of method: org.apache.commons.logging.impl.SLF4JLog.deub() is applicable for argument types: (java.lang.String) values: [using the default translation table]

```

**Comment by deepakunni3** [comment]
Few issues:
1. 'Non-default Translation Table' value specified in the Organisms Panel **DOES NOT** persist after adding an organism. But 'Non-default Translation Table' value **DOES** persist after updating for an existing organism via the Organisms Panel.
2. There are methods in `FeatureService` that directly call `configWrapperService.getTranslationTable()` instead of checking for the organism's non-default translation table.
3. There are methods in `SequenceService` that directly call `configWrapperService.getTranslationTable()` instead of checking for the organism's non-default translation table

**Comment by nathandunn** [comment]
thanks @deepakunni3   I'll clean those up.

**Comment by nathandunn** [comment]
@deepakunni3  I think its ready for a final review.    sorry there was so much junk in there.  

I also ended up collapsing away some methods we don't really use.

**Comment by deepakunni3** [comment]
@nathandunn Works like a charm! 👍 

Ready to merge.

**Pull request merged**